File: app/src/main/java/apps/gliger/glg/lar/Constant/SubjectDiffCallback.java
package apps.gliger.glg.lar.Constant;

import android.support.annotation.Nullable;
import android.support.v7.util.DiffUtil;

import java.util.List;

import apps.gliger.glg.lar.RoomDB.Subject;

public class SubjectDiffCallback extends DiffUtil.Callback {

    private final List<Subject> oldSubList;
    private final List<Subject> newSubList;

    public SubjectDiffCallback(List<Subject> oldSubList, List<Subject> newSubList) {
        this.oldSubList = oldSubList;
        this.newSubList = newSubList;
    }

    @Override
    public int getOldListSize() {
        return oldSubList.size();
    }

    @Override
    public int getNewListSize() {
        return newSubList.size();
    }

    @Override
    public boolean areItemsTheSame(int i, int i1) {
        return oldSubList.get(i).getId()==newSubList.get(i1).getId();
    }

    @Override
    public boolean areContentsTheSame(int i, int i1) {
        return oldSubList.get(i).getSubjectName().equals(newSubList.get(i1).getSubjectName());
    }

    @Nullable
    @Override
    public Object getChangePayload(int oldItemPosition, int newItemPosition) {
        return super.getChangePayload(oldItemPosition, newItemPosition);
    }
}

File: app/src/main/java/apps/gliger/glg/lar/Constant/SubjectDiffCallback.kt
package apps.gliger.glg.lar.Constant

import android.support.v7.util.DiffUtil

import apps.gliger.glg.lar.RoomDB.Subject

class SubjectDiffCallback(private val oldSubList: List<Subject>, private val newSubList: List<Subject>) : DiffUtil.Callback() {

    override fun getOldListSize(): Int {
        return oldSubList.size
    }

    override fun getNewListSize(): Int {
        return newSubList.size
    }

    override fun areItemsTheSame(i: Int, i1: Int): Boolean {
        return oldSubList[i].id == newSubList[i1].id
    }

    override fun areContentsTheSame(i: Int, i1: Int): Boolean {
        return oldSubList[i].subjectName == newSubList[i1].subjectName
    }

    override fun getChangePayload(oldItemPosition: Int, newItemPosition: Int): Any? {
        return super.getChangePayload(oldItemPosition, newItemPosition)
    }
}
